distribution/lib/Standard/Table/0.2.32-SNAPSHOT/src/Data/Matching.enso

```diff
@@ -0,0 +1,167 @@
+from Standard.Base import all
+import Standard.Base.Data.Locale
+import Standard.Base.Data.Text.Regex as Regex_Module
+
+from Standard.Base.Error.Problem_Behavior as Problem_Behavior_Module import all
+
+## Strategy for matching names.
+type Matching_Strategy
+   ## UNSTABLE
+      Exact name matching.
+
+      A name is matched if its exact name is provided.
+   type Exact (case_sensitivity : (True | Case_Insensitive) = True)
+
+   ## UNSTABLE
+      Regex-based name matching.
+
+      A name is matched if its name matches the provided regular expression.
+   type Regex (case_sensitivity : (True | Case_Insensitive) = True)
+
+## UNSTABLE
+   Specifies that the operation should ignore case.
+
+   TODO: Since case-sensitive can be locale dependent, in the future this may
+         be extended with a `locale` setting.
+type Case_Insensitive
```

Comment:
Right now, the `Case_Insensitive` atom does not have any options, so `True | Case_Insensitive` is not better than `Bool` in any way. Until it has options, I don't see any reason to keep it this way TBH.

Reply:
The intention is to work on locale insensitive over the next week as it is part of the API for the Select libraries. 

This felt a reasonable placeholder so the base function could be built on top of.

Reply:
Also the idea is forward compatibility - if we do `True | False` we need to break old APIs if we want to extend it.

If we keep `True | Case_Insensitive` we can add default parameters to `Case_Insensitive`, so that `Case_Insensitive` from old code still works (falling back to the sane defaults which would preserve the old semantics).